kombu/asynchronous/semaphore.py

```diff
@@ -93,8 +100,8 @@ def __repr__(self):
 class DummyLock:
     """Pretending to be a lock."""
 
-    def __enter__(self):
+    def __enter__(self) -> 'DummyLock':
         return self
 
-    def __exit__(self, *exc_info):
+    def __exit__(self, *exc_info: Any) -> None:
```

Comment:
https://sethmlarson.dev/blog/tests-arent-enough-case-study-after-adding-types-to-urllib3#anything-but-any

Reply:
We can replace exc_info by the three parameters and annotate them properly.

Reply:
I'll replace it with this 👍 
```python
def __exit__(self, exc_type: type, exc_val: Exception, exc_tb: 'TracebackType') -> None:
```